File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        st=[]
        maxi=0
        i=0
        left=0
        setS=set(s)
        while i<len(s):
            if s[i] not in st:
                st.append(s[i])
                i+=1
                maxi=max(maxi,len(st))
                if maxi==len(setS):
                    break
            else:
                while s[left]!=s[i]:
                    left+=1
                if s[i] in st:
                    st.pop(0)
                
        return maxi

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int maxi=0,count=0,left=0,right=0,i=0;
        vector<int> m(256,-1);
        while(right<s.size()){
            if(m[s[right]]!=-1) left=max(m[s[right]]+1,left);
            m[s[right]]=right;
            maxi=max(right-left+1,maxi);
            right++;
            
//             if(m.find(s[i])!=m.end()){
//                 count=1;
//                 maxi=max(count,maxi);
//                 m.clear();
//                 m[s[i]]=1;
//             }else{
//                 m[s[i]]=1;
//                 count++;
                
//             }
            cout<<"---"<<left<<right<<endl;
            // maxi=max(maxi,count);
        }
        return maxi;
    }
};